src/iotconnect_telemetry.c

```diff
@@ -25,15 +25,21 @@ struct IotclMessageHandleTag {
 };
 
 cJSON *json_object_dotset_locate(cJSON *search_object, char **leaf_name, const char *path) {
+    *leaf_name = NULL;
+
     static const char *DELIM = ".";
     char *mutable_path = iotcl_strdup(path);
+    if (NULL == mutable_path) {
+        return NULL;
+    }
+
     char *token = strtok(mutable_path, DELIM);
-    *leaf_name = NULL;
```

Comment:
So that it you return from the function it is set to NULL.

New C standards mean that declarations don't have to be at the top of the function.
One of the declarations here invokes a function.